src/bloqade/squin/rewrite/canonicalize.py

```diff
@@ -58,3 +62,68 @@ def rewrite_Statement(self, node: ir.Statement) -> abc.RewriteResult:
 
         node.delete()
         return abc.RewriteResult(has_done_something=True)
+
+
+@dataclass
+class CanonicalizeHermitian(abc.RewriteRule):
+    hermitian_values: dict[ir.SSAValue, HermitianLattice]
+
+    def rewrite_Statement(self, node: ir.Statement) -> abc.RewriteResult:
+        if not isinstance(node, op.stmts.Operator):
+            return abc.RewriteResult()
+
+        maybe_hermitian = node.get_trait(op.traits.MaybeHermitian)
+
+        if maybe_hermitian is None:
+            return abc.RewriteResult()
+
+        is_hermitian = self.hermitian_values.get(node.result)
+        if is_hermitian is None:
+            return abc.RewriteResult()
+
+        if is_hermitian.is_equal(Hermitian()):
+            maybe_hermitian.set_hermitian(node, True)
+        else:
+            maybe_hermitian.set_hermitian(node, False)
+
+        return abc.RewriteResult(has_done_something=True)
+
+
+@dataclass
+class CanonicalizeUnitary(abc.RewriteRule):
+    unitary_values: dict[ir.SSAValue, UnitaryLattice]
+
+    def rewrite_Statement(self, node: ir.Statement) -> abc.RewriteResult:
+        if not isinstance(node, op.stmts.Operator):
+            return abc.RewriteResult()
+
+        maybe_unitary = node.get_trait(op.traits.MaybeUnitary)
+
+        if maybe_unitary is None:
+            return abc.RewriteResult()
+
+        is_unitary = self.unitary_values.get(node.result)
+        if is_unitary is None:
+            return abc.RewriteResult()
+
+        if is_unitary.is_equal(Unitary()):
+            maybe_unitary.set_unitary(node, True)
+        else:
+            maybe_unitary.set_unitary(node, False)
```

Comment:
```suggestion
        new_unitary_status = is_unitary.is_equal(Unitary())
        maybe_unitary.set_unitary(node, new_unitary_status)
```
Just feels a little tidier to me